scripts/calculateRevenue/protocols/somm/index.ts

```diff
@@ -0,0 +1,123 @@
+import { Address, erc20Abi, formatUnits, getContract, isAddress } from 'viem'
+import { getViemPublicClient } from '../../../utils'
+import { getVaults } from './getVaults'
+import { VaultInfo } from './types'
+import { getEvents } from './getEvents'
+
+const ONE_DAY = 24 * 60 * 60 * 1000
+
+export async function getBalanceOfAddress({
+  vaultInfo,
+  address,
+}: {
+  vaultInfo: VaultInfo
+  address: Address
+}) {
+  const client = getViemPublicClient(vaultInfo.networkId)
+  const vaultContract = getContract({
+    address: vaultInfo.vaultAddress,
+    abi: erc20Abi,
+    client,
+  })
+  return vaultContract.read.balanceOf([address])
+}
+
+/**
+ * Calculates the daily mean Total Value Locked (TVL) for a given user address
+ * and vault pair within a specified time range.
+ *
+ * TODO(ENG-201): Return TVL in USD
+ */
+export async function getDailyMeanTvl({
+  vaultInfo,
+  address,
+  startTimestamp,
+  endTimestamp,
+  nowTimestamp,
+}: {
+  vaultInfo: VaultInfo
+  address: Address
+  startTimestamp: Date
+  endTimestamp: Date
+  nowTimestamp: Date
+}) {
+  if (endTimestamp.getTime() > nowTimestamp.getTime()) {
+    throw new Error('Cannot have an endTimestamp in the future')
+  }
+  const client = getViemPublicClient(vaultInfo.networkId)
+  const vaultContract = getContract({
+    address: vaultInfo.vaultAddress,
+    abi: erc20Abi,
+    client,
+  })
+  const currentLPTokenBalance = await vaultContract.read.balanceOf([address])
+  const tokenDecimals = await vaultContract.read.decimals()
+
+  const tvlEvents = await getEvents({
+    address,
+    vaultInfo,
+    startTimestamp,
+    endTimestamp: nowTimestamp,
```

Comment:
just to make sure I understood this correctly. We fetch until now instead of endTimestamp so that if any event happened afterwards, we adjust the balance correctly. We just won't be counting them because `daysInRange` will be 0 for those events right?

Reply:
correct